central/authprovider/service/service_impl.go

```diff
@@ -186,6 +192,14 @@ func (s *serviceImpl) PutAuthProvider(ctx context.Context, request *storage.Auth
 	// Attempt to merge configs.
 	request.Config = provider.MergeConfigInto(request.GetConfig())
 
+	// Verify whether there were any changes made to the provider's required attributes. This is currently not allowed
```

Comment:
I'm not sure we need this check because changes to active auth providers are not allowed anyway (enforced by `registry.DeleteProvider()` below.

Reply:
While that's true, it could be the case that users add another identity provider and change the RH SSO provider, there's nothing stopping them from doing that. I'd rather be on the defensive side with this one and make sure we cover those cases.

Reply:
So what you say is that if for whatever reason we ship with several auth providers so that users can activate the non-RHSSO one, log in via it, and then eliminate the required attribute check effectively letting every RH user to their instance. I didn't think about that, but it sounds like they should better not do that for their own benefit : ).

I'm also not sure how to create not `active` auth providers. If we can ensure our auth provider is `active` and hence can't be changed, we might just rely on that.